Rubberduck.Parsing/UIContext/UiSynchronizer.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Rubberduck.Parsing.UIContext
+{
+    public static class UiSynchronizer
```

Comment:
I suspect we might eventually want to call other exposed VBE dll functions, and the structure for detecting Vbe6/Vbe7 could be reused.  Perhaps consider renaming to VbeDllAccessor, or something along those lines.

Reply:
At this point, I'm gambling that those will be the only functions we need. If it's not, then it will definitely have to be split into its own class solely responsible for exposing VBE's exported functions.

Reply:
I'd bet the other way... come back in a few years and we'll find out :+1:

Reply:
Perhaps it's just me, but the name 'UiSynchronizer' still doesn't sit well with me.  It appears the purpose of this class is to call the VBE exposed DoEvents function, with a guard to ensure we only call it from the main thread.   We're not really syncing anything?

Reply:
My word. I think something in the direction of `ComMessagePumper` would be more appropriate.

Reply:
What I didn't mention was that I envisioned that in a sequent PR, the class would provide mechanism for actually synchronizing using something like `ReaderWriterLockSlim` to help manage the COM access. Should this be made into a dedicated class responsible for pumping message solely?

Reply:
@MDoerner what's with the 'my word' attitude?  Naming stuff is hard.   Important to get it right, for the sake of those who come after you.

Reply:
@bclothier ... I would.  up to you of course.

Reply:
@WaynePhillipsEA I guess this is a language issue. Read it as _I fully agree._.

Reply:
@MDoerner in which case, I apologize. I read it like 'oh my word', which over here at least, and in context, would mean along the lines of 'why the f*ck are we even nitpicking at this'.